CMakeLists.txt

```diff
@@ -187,10 +187,11 @@ if(NOT CONFIG_RTTI)
   )
 endif()
 
-if(NOT CONFIG_NATIVE_APPLICATION)
-  set(NOSTDINC_F -nostdinc)
+if (NOT CONFIG_NEWLIB_LIBC AND NOT CONFIG_NATIVE_APPLICATION)
+toolchain_cc_nostdinc()
```

Comment:
I know the name reveals the meaning here, but for consistency, we should document the intent of these macros.
nit: indent.